wsd/RequestVettingStation.cpp

```diff
@@ -353,7 +353,8 @@ bool RequestVettingStation::transferToDocBroker(const std::string& url,
     if (std::shared_ptr<DocumentBroker> docBroker = createDocBroker(docKey, configId, url, uriPublic))
     {
         launchInstallPresets();
-        bTransferred = createClientSession(disposition, ws, docBroker, docKey, url, uriPublic);
+        createClientSession(disposition, ws, docBroker, docKey, url, uriPublic);
```

Comment:
I think createDocBroker followed by createClientSession is fundamentally unsafe. The DocBroker can have it's poll running already owing to races and - then we do createClientSession from different threads creating problems...

Reply:
RequestVettingStation::createClientSession now moves all of its "real" work into the DocumentBroker::setupTransfer callback so DocumentBroker::createNewClientSession is called by its polling thread, which I think is what we want here.